tile.js

```diff
@@ -12,24 +12,28 @@ class Tile {
     for (let i = 0; i < tiles.length; i++) {
       let tile = tiles[i];
       // UP
-      if (tile.edges[2] == this.edges[0]) {
+      if (this.reverseEdge(tile.edges[2]) == this.edges[0]) {
         this.up.push(i);
       }
       // RIGHT
-      if (tile.edges[3] == this.edges[1]) {
+      if (this.reverseEdge(tile.edges[3]) == this.edges[1]) {
         this.right.push(i);
       }
       // DOWN
-      if (tile.edges[0] == this.edges[2]) {
+      if (this.reverseEdge(tile.edges[0]) == this.edges[2]) {
         this.down.push(i);
       }
       // LEFT
-      if (tile.edges[1] == this.edges[3]) {
+      if (this.reverseEdge(tile.edges[1]) == this.edges[3]) {
         this.left.push(i);
       }
     }
   }
 
+  reverseEdge(edge) {
+    return edge.split("").reverse().join("");
```

Comment:
We now compare each tile side label against the mirrored image of itself. For symmetrical tiles it's "unnecessary" as it compares '000' with '000'. But for asymmetrical tiles it compares successfully '100' against the right match '001'.

This auxiliary method just reverses the string by transforming it first to an array to use the `reverse()` method.